libraries/AP_HAL_Linux/Storage.cpp

```diff
@@ -26,6 +26,8 @@ using namespace Linux;
 #define STORAGE_DIR "/var/lib/ardupilot"
 #elif APM_BUILD_TYPE(APM_BUILD_Replay)
 #define STORAGE_DIR "."
+#elif CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_RST_ZYNQ
+#define STORAGE_DIR "/mnt/APM"
```

Comment:
is this differentiation needed? we are trying to standardize boards on `/var/lib/ardupilot`

Reply:
yes, our board provide both ubuntu and ramdisk filesystem. In ramdisk, log data have to be saved in  sd card, whose path is /mnt/

Reply:
if you can bind mount directories, then that's a non-issue... it's just a matter of bind-mounting e.g. /mnt/ardupilot to the "correct" place.... And you can do that before starting ardupilot. In a systemd's .service file it would be a matter of adding a `ExecPre` line or adding it to fstab.

I won't enforce this though, so your call what you want to do.